sklearn/decomposition/_pca.py

```diff
@@ -358,6 +360,26 @@ class PCA(_BasePCA):
     [6.30061...]
     """
 
+    _parameter_constraints = {
+        "n_components": [
+            Interval(Integral, 0, None, closed="left"),
+            Interval(Real, 0, 1, closed="neither"),
+            StrOptions({"mle"}),
+            None,
+        ],
+        "copy": [bool],
+        "whiten": [bool],
+        "svd_solver": [StrOptions({"auto", "full", "arpack", "randomized"})],
+        "tol": [Interval(Real, 0, None, closed="left")],
+        "iterated_power": [
+            StrOptions({"auto"}),
+            Interval(Integral, 0, None, closed="left"),
+        ],
+        "n_oversamples": [Interval(Integral, 1, None, closed="left")],
+        "power_iteration_normalizer": [StrOptions({"auto", "QR", "LU", "none"})],
+        "random_state": ["random_state"],
```

Comment:
shouldn't the random state be an int, int, RandomState instance or None?

Reply:
it is, but the string "random_state" is a convenience tool to not have to repeat that everywhere. If you're interested here's where it's defined https://github.com/scikit-learn/scikit-learn/blob/79c176de3f8a6972fe9e087f612f77bbd2b40ad6/sklearn/utils/_param_validation.py#L394

Reply:
Thanks for answering. Clear enough.